### ISSUE TITLE
QUESTION: How to set SSH_AUTH_SOCK

### ISSUE THREAD

**Issue by aradapilot** [beekeeper-studio/beekeeper-studio#444]
**Details:**
Trying to set SSH_AUTH_SOCK to be able to use ssh-agent properly.  in general, this will work for any application:
SSH_AUTH_SOCK="/path/to/correct/sock" /path/to/binary
but it doesn't here, I get "We found your ssh agent, you're good to go!" but any connection attempt errors with "All configured authentication methods failed" - there's no more detail than this, but I never get the hardware prompts signalling the correct sock file is in use, so that seems to be the issue.
Debug mode doesn't output anything from ssh, so that doesn't add much.  It also doesn't output the contents of platformInfo (from https://github.com/beekeeper-studio/beekeeper-studio/blob/master/src/common/platform_info.ts#L21), so I have no way to see what it's using internally.  It *does* output env, and shows SSH_AUTH_SOCK set in the env successfully, but that was never really in doubt.
So - how do I set it?  Or is there something else I'm missing?  More debug output and detail in error messages would definitely be useful, though that's a separate issue.

**Version Information [might be relevant to your issue]**
Operating System: Ubuntu & mac
App Version [help -> about]: 1.8.7

**Comment by rathboma** [comment]
So if you have your SSH agent running, and you've authenticated your SSH keys then it should just work.

Have you run `ssh-add` to add a key to your ssh agent keychain?

**Comment by aradapilot** [comment]
Yeah, it's in there.  The issue is, instead of a passphrase to unlock, it's a hardware cue (see yubico/yubikey for details) - which is handled by most things gracefully if you change the env variable, but that doesn't seem possible here.

**Comment by kitsunet** [comment]
Seems I am in the same boat, happy to provide any needed info. I run the gpg-agent (a replacement ssh-agent) to authenticate via yubikey and while I also get the "found ssh agent" message any connection attempt runs into the same error as above.

**Comment by rathboma** [comment]
I'm lacking a Yubikey, or any other hardware device, and I don't really understand the complexities of what to do differently.

Could someone give me a quick step-by-step of what the app should do differently? Like do I need to do something with the SSH_AUTH_SOCK value or something?

**Comment by gyamada619** [comment]
In my case the issue seems to be that Beekeeper is not sending the `SendEnv` parameter that's in my `.ssh/config` file.

**Comment by hlidotbe** [comment]
I have the same issue. I assume it's because I set the SSH_AUTH_SOCK in my .zshrc which is not used by the desktop environment and so it uses the system agent and not the socket created by yubikey-agent
